### PR TITLE
[WIP] Fix ref counting

### PR DESCRIPTION
**Pull request by xVin**
targets #479

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Note: Should be merged after almost all other PRs, as these may introduce code which needs to be fixed by this PR, too.

File: src/serialize/json/serializer.c [hunk 1]
```diff
@@ -466,8 +466,9 @@ serialize_value(
             struct ws_string* str;
             str = ws_value_string_get((struct ws_value_string*) val);
 
-            char* buf = ws_string_raw(str);
-
+            char* buf = ws_string_raw(str); //!< buf is a copy
```
Comment (matthiasbeyer) [inline src/serialize/json/serializer.c]: Please don't use `//!<` here, but `//`

Comment (xVin) [inline src/serialize/json/serializer.c]: oops

**Comment by neithernut** [comment]
ACK.
I'll merge it right away, so it won't get stalled again.

**Comment by xVin** [comment]
Er, this was still WIP, but shouldnt be much missing, I'll just submit a small addon tomorrow.

**Comment by neithernut** [comment]
I know that it was WIP, but I really did want to avoid stalling this PR again, as it brought fixes with it.

**Pull request merged**